src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1180,11 +1200,22 @@ void MarkEntireAssembly (AssemblyDefinition assembly)
 
 			foreach (TypeDefinition type in assembly.MainModule.Types)
 				MarkEntireType (type, includeBaseTypes: false, new DependencyInfo (DependencyKind.TypeInAssembly, assembly), null);
+
+			ProcessEmbeddedXml (assembly);
+		}
+
+		void ProcessEmbeddedXml (AssemblyDefinition assembly)
```

Comment:
This is outdated and has been removed.

Based on this [discussion](https://github.com/mono/linker/pull/1666#discussion_r544475617) it doesn't make sense to process all XML in the same place. The descriptors should only be processed if the assembly is marked, whereas substitutions/attributes might be needed even if not marked. For example we might ask whether an attribute has `RemoveAttributeInstances`, when the attribute is defined in an assembly that's never marked.

If we need a per-assembly pipeline concept, the best place to do it might be here: https://github.com/sbomer/linker/blob/aa4728ce65768b4067c275694cad630c5efa42e8/src/linker/Linker.Steps/MarkStep.cs#L1276-L1286 - this is called when an assembly is marked - but it doesn't include all of the XML steps.

Reply:
As discussed, the steps which run lazily on marked assemblies will be hard-coded for now. A separate pipeline may be introduced as part of https://github.com/mono/linker/issues/1735.